tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -258,27 +340,25 @@ ansible_winrm_server_cert_validation=ignore`, ip, password, winRMPort))
 
 // testAnsiblePing checks if ansible is able to ping on opened winRmHttps port
 func testAnsiblePing(t *testing.T) {
-	// this regex looks for the IP address pattern from vmCredentialPath.
-	// the sample vmCredentialPath looks like xfreerdp /u:xxxx /v:xx.xx.xx.xx /h:1080 /w:1920 /p:'password1234'
-	ipAddressPattern := regexp.MustCompile(`\d+.\d+.\d+.\d+`)
-	// the passwordPattern extracts the characters present after the '/p:', but we are extracting
-	// 13 characters even though the windows password length is of size 12 because we got a single quote
-	// character included in the password.
-	passwordPattern := regexp.MustCompile(`/p:.{13}`)
+	// TODO: Do not iterate on the instances but instead pass the credentials object
+	//  for test function
 	for _, vmName := range instanceIDs {
-		vmCredentialPath := filepath.Join(dir, "/", vmName)
-		rdpCmd, err := ioutil.ReadFile(vmCredentialPath)
+		ipAddress, password, err := getIpPass(vmName)
 		require.NoErrorf(t, err, "failed to read file %s", vmName)
-		ipAddress := ipAddressPattern.FindString(string(rdpCmd))
 		assert.NotEmpty(t, ipAddress, "the IP address can't be empty")
-		password := passwordPattern.FindString(string(rdpCmd))[3:]
 		assert.NotEmpty(t, password, "the password can't be empty")
-		// we are trimming out the unnecessary single quotes.
-		password = strings.Trim(password, `'`)
 		hostFileName, err := createHostFile(ipAddress, password)
 		require.NoError(t, err, "failed to create a temp file")
 		pingCmd := exec.Command("ansible", "win", "-i", hostFileName, "-m", "win_ping")
 		out, err := pingCmd.CombinedOutput()
 		assert.NoErrorf(t, err, "ansible ping check failed with error: %s", string(out))
 	}
 }
+
+// testAzureFirewallRule asserts if the created instance has firewall rule that opens container logs port.
+func testAzureInstancesFirewallRule(t *testing.T, creds types.Credentials) {
+	ipAddress, password, err := getIpPass(creds.GetInstanceId())
+	require.NoErrorf(t, err, "failed to obtain credentials for %s", creds.GetInstanceId())
+	credentials = types.NewCredentials(creds.GetInstanceId(), ipAddress, password, azureUser)
```

Comment:
Why are you doing this again? Won't calling `getCredentialObject()` in `setup()` suffice?

Reply:
good catch aravindh!, will remove the creds argument from the test function.